tests/channel_testing/reposerver.py

```diff
@@ -343,11 +343,14 @@ def do_GET(self):
     handler = BasicAuthHandler
 elif args.auth == "token" or args.token:
     handler = CondaTokenHandler
+else:
+    print("No auth method given.")
+    exit(1)
 
 PORT = args.port
 
 server = HTTPServer(("", PORT), handler)
-print("Server started at localhost:" + str(PORT))
+print("Server started at localhost:" + str(server.server_port))
```

Comment:
Updating this to allow an OS-chosen port (ask for port 0, and the OS chooses an available port).